lib/gemstash/http_client.rb

```diff
@@ -35,12 +35,17 @@ def self.for(upstream)
         config.adapter :net_http
         config.options.timeout = gemstash_env.config[:fetch_timeout]
       end
+
+      client.basic_auth(upstream.user, upstream.password) if upstream.auth?
```

Comment:
Minor: perhaps we can check if the basic_autj method is defined, before using it. There's been a new version, which has another variant method. Faraday  has a nite about that in its website.

Reply:
A note.

Reply:
I can definitely add that if it's warranted. My thinking was this kind of check was unnecessary since the gemspec is locking the Faraday gem to `~> 0.9` and the new interface only exist in `>= 1.0`, and unit test coverage will show that the major version bump of Faraday will break existing functionality and force the necessary code change.

Reply:
Cool, cool, we'll get the relevant deprecation warnings when we get there.